packages/ember-data/lib/system/model/model.js

```diff
@@ -887,8 +599,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
       and value is an [oldProp, newProp] array.
   */
   changedAttributes: function() {
-    var oldData = get(this, '_data');
-    var newData = get(this, '_attributes');
+    var oldData = get(this.reference, '_data');
```

Comment:
it seems like this logic should move into the reference object rather than reaching into its private api.

something like `Reference.prototype.changedAttributes`. Then this function becomes `this.reference.changedAttributes()`